diff --git a/scripts/headers_install.sh b/scripts/headers_install.sh
index dd554bd436cc..110d8e2cbb5c 100755
--- a/scripts/headers_install.sh
+++ b/scripts/headers_install.sh
@@ -29,7 +29,7 @@ if [ -n "$(sed -n -e "/SPDX-License-Identifier:.*GPL-/{/WITH Linux-syscall-note/
 	exit 1
 fi
 
-sed -E -e '
+sed -r -e '
 	s/([[:space:](])(__user|__force|__iomem)[[:space:]]/\1/g
 	s/__attribute_const__([[:space:]]|$)/\1/g
 	s@^#include <linux/compiler(|_types).h>@@
diff --git a/scripts/link-vmlinux.sh b/scripts/link-vmlinux.sh
index 6ae0d70089b4..407ca3825710 100755
--- a/scripts/link-vmlinux.sh
+++ b/scripts/link-vmlinux.sh
@@ -219,7 +219,7 @@ gen_btf()
 		return 1
 	fi
 
-	pahole_ver=$(${PAHOLE} --version | sed -E 's/v([0-9]+)\.([0-9]+)/\1\2/')
+	pahole_ver=$(${PAHOLE} --version | sed -r 's/v([0-9]+)\.([0-9]+)/\1\2/')
 	if [ "${pahole_ver}" -lt "116" ]; then
 		echo >&2 "BTF: ${1}: pahole version $(${PAHOLE} --version) is too old, need at least v1.16"
 		return 1
